### PR TITLE
Fix formatting ActionController::Parameters

### PR DESCRIPTION
**Pull request by HarlemSquirrel**
Add an extra check for unpermitted Rails paramters
that will report they they respond to :to_hash
but raise an error.

Fixes #17

### CODE REVIEW

File: spec/ext/active_record_spec.rb [hunk 1]
```diff
@@ -259,7 +259,7 @@ class SubUser < User {
           expect(out).to match(
             /\s*first\(\*args,\s+&block\)\s+#<Class:ActiveRecord::Base> \(ActiveRecord::Querying\)/
           )
-        elsif RUBY_VERSION >= '2.4.4'
+        elsif RUBY_VERSION =~ /2\.[4-6]\.([4-9]|[1-9][0-9])/
```
[the remark this answers is not in the document]
Comment (HarlemSquirrel) [reply spec/ext/active_record_spec.rb:262]: So we need to find version greater than or equal to 2.4.4 and now that 2.4.10 is out, the string comparison fails with that version.

```rb
'2.4.3' >= '2.4.4'
# => false
'2.4.4' >= '2.4.4'
# => true
'2.4.10' >= '2.4.4'
# => false

'2.4.3' =~ /2\.[4-6]\.([4-9]|[1-9][0-9])/
# => nil
'2.4.4' =~ /2\.[4-6]\.([4-9]|[1-9][0-9])/
# => 0
'2.4.10' =~ /2\.[4-6]\.([4-9]|[1-9][0-9])/
# => 0
```

[the remark this answers is not in the document]
Comment (paddor) [reply spec/ext/active_record_spec.rb]: I see. But the restrictions then also apply to Ruby 2.5 and 2.6.

How about: `/^2\.4\.([4-9]|[1-9][0-9])|^2\.[56]\./`

[the remark this answers is not in the document]
Comment (HarlemSquirrel) [reply spec/ext/active_record_spec.rb]: Updated.

**Comment by HarlemSquirrel** [comment]
OK, I had to run `bundle exec appraisal rails-6.0 rspec` to reproduce this new failure locally. It appears that with Ruby 2.7 and ActiveRecord 6.0.3 things have changed a bit. I have to see what's the best approach to handle this.

**Pull request merged**